hoomd/mpcd/DryingDropletStreamingMethod.h

```diff
@@ -0,0 +1,87 @@
+#include "RandomSubsetPicker.h"
+#include "SphereGeometry.h"
+
+namespace mpcd
+{
+//! MPCD DryingDropletStreamingMethod
+/*!
+ * This method implements the base version of ballistic propagation of MPCD
+ * particles in drying droplet.
+ *
+ * The integration scheme is essentially Verlet with specular reflections.First the radius and Velocity of droplet is updated then
+ * the particle is streamed forward over the time interval. If the particle moves outside the Geometry(spherical droplet), it is placed back
+ * on the boundary and particle velocity is updated according to the boundary conditions.Streaming then continues and
+ * place the particle inside droplet(sphereGeometry). And particle which went outside or collided with droplet interface was marked.
+ * Right amount of marked particles are then evaporated.
+ *
+ * To facilitate this, ConfinedStreamingMethod must flag the particles which were bounced back from surface.
+ */
+
+class PYBIND11_EXPORT DryingDropletStreamingMethod : public mpcd::ConfinedStreamingMethod<mpcd::detail::SphereGeometry>
+    {
+    public:
+        //! Constructor
+        /*!
+         * \param sysdata MPCD system data
+         * \param cur_timestep Current system timestep
+         * \param period Number of timesteps between collisions
+         * \param phase Phase shift for periodic updates
+         * \param R is the radius of sphere
+         * \param bc is boundary conditions(slip or no-slip)
+         * \param density is solvent number density inside the droplet
+         * \param seed is Seed to random number Generator
+         */
+        DryingDropletStreamingMethod(std::shared_ptr<mpcd::SystemData> sysdata,
+                                    unsigned int cur_timestep,
+                                    unsigned int period,
+                                    int phase,
+                                    std::shared_ptr<::Variant> R,
+                                    mpcd::detail::boundary bc,
+                                    Scalar density,
+                                    unsigned int seed);
+
+        //! Implementation of the streaming rule
+        virtual void stream(unsigned int timestep);
+
+    protected:
+        std::shared_ptr<::Variant> m_R;                    //!< Radius of Sphere
+        mpcd::detail::boundary m_bc;                       //!< Boundary condition
+        Scalar m_density;                                  //!< Solvent density
+        unsigned int m_seed;                               //!< Seed to evaporator pseudo-random number generator
+
+        GPUArray<unsigned int> m_picks;                   //!< Indices of Particles picked for evaporation on this rank in /a m_bounced array
+        GPUVector<mpcd::detail::pdata_element> m_removed;  //!< Hold output particles that are removed
+
+    private:
+        RandomSubsetPicker m_picker;                       //!< For randomly picking particles to evaporate from particles which are bounced
```

Comment:
These should all be either private or protected, we can go with protected:
```suggestion
        RandomSubsetPicker m_picker;                       //!< Picker to remove particles
        GPUArray<unsigned int> m_picks;                   //!< Indexes of particles to remove
        GPUVector<mpcd::detail::pdata_element> m_removed;  //!< Removed particle data (not used after removal)
```